cli/tests/unit/serve_test.ts

```diff
@@ -2826,6 +2828,102 @@ Deno.test(
   },
 );
 
+Deno.test(
+  "Throw if disturbed",
+  { permissions: { net: true } },
+  async function shouldThrowIfBodyIsUnusableDisturbed() {
+    const ac = new AbortController();
+    const listeningPromise = deferred();
+
+    function customErrorCb(ac: AbortController): (error: unknown) => Response {
+      return (error) => {
+        assert(error instanceof TypeError);
+        assert(
+          error.message.endsWith(
+            "Body is unusable.",
+          ),
+        );
+                  
+        ac.abort();
+        return new Response("Internal server error", { status: 500 });
+      };
+    }    
+
+    const server = Deno.serve({
+      handler: async (req) => {
+        await req.text();
+
+        req.clone();
```

Comment:
I would suggest adding try/catch around these blocks, and asserting if it succeeds like so:

```
try {
  req.clone();
  fail("This should not have succeeded");
} catch (e) {
  assert(...)
}
```